webapplication/aoi/views.py

```diff
@@ -174,8 +185,27 @@ def create(self, request, *args, **kwargs):
         if not component.validated and \
                 not self.request.user.has_perm('aoi.can_run_not_validated'):
             return Response(status=status.HTTP_403_FORBIDDEN)
+
         serializer.is_valid(raise_exception=True)
-        self.perform_create(serializer)
+        request_price = component.calculate_request_price(
+            user=request.user,
+            area=serializer.validated_data["aoi"].area_in_sq_km
+        )
+        user_actual_balance = request.user.actual_balance
+        if user_actual_balance < request_price:
+            return Response(_(f"Your actual balance is {request.user.actual_balance}. "
+                              f"It’s not enough to run the request. Please replenish the balance. "
+                              f"Contact support (support@soilmate.ai)"),
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This should be in a non_field_errors, see https://github.com/QuantuMobileSoftware/geoap/blob/master/webviewer/src/api/instance.js and https://docs.djangoproject.com/en/4.1/ref/forms/api/#django.forms.Form.non_field_errors